src/rubber.ts

```diff
@@ -59,14 +59,33 @@ export interface IRubberOptions {
 export function compile(options: IRubberOptions) {
     const emitter = new EventEmitter() as RubberEventEmitter; // we dont need the overhead of a sub class
     const projectFile = resolve(options.projectPath);
+    const platform = options.platform;
 
     // Build component for checking later.
-    const component = "Windows";
-    const componentBuild = "Windows.build_module";
+    let component = "Windows";
+    let componentBuild = "Windows.build_module";
+    let defaultPackageKey = "Package";
+    switch (platform){
+        case "android":
+            component = "Android";
+            componentBuild = "android.build_module";
+            break;
+        case "switch":
+            component = "Switch";
+            componentBuild = "switch.build_module";
+            break;            
+        default:
+            defaultPackageKey = "PackageZip";
+            component = "Windows";
+            componentBuild = "Windows.build_module";
+            break;            
+    }
+
+
     // We want to run stuff async with await, so this will be in its own function.
     const asyncRun = async() => {
         // !!! Other platforms support
-        if(options.platform !== "windows") throw new Error("Cannot compile to platform '" + options.platform + "'");
+        //if(options.platform !== "windows") throw new Error("Cannot compile to platform '" + options.platform + "'");
```

Comment:
This part worries me a little because trying out other platforms will default to windows which may be really unexpected. If you want to you can fix it or ill fix it in about a day.

Reply:
Fair point! I will change the Switch statement default to an error value, which this statement will catch. That way, if the user tries unsupported platforms, it will give the error instead of defaulting to Windows.